src/Microsoft.OData.Edm/Csdl/Parsing/Common/EdmXmlDocumentParser.cs

```diff
@@ -407,40 +407,64 @@ protected string ValidateEnumMembersPath(string path)
 
         private string ValidateTypeName(string name)
         {
-            string[] typeInformation = name.Split(new char[] { '(', ')' });
-            string typeName = typeInformation[0];
+            Debug.Assert(name != null, "name != null");

```

Comment:
If it's validation logic, so 'null' is invalid and should throw exception, right?

Reply:
@xuzhg I could rename it `CheckAndReportTypeNameValidity`. I think that's intuitive enough.
However, I'd want you to note that all other methods that do something similar in the class are named `Validate*`. I'll leave them as is...
![image](https://github.com/user-attachments/assets/82430e18-8b0f-45b5-9de6-53497172a15a)